### ISSUE TITLE
rpminspect: archive_read_next_header(): No such file or directory

### ISSUE THREAD

**Issue by morucci** [rpminspect/rpminspect#308]
Hi David,

I saw that rpminspect crashed during grub2 packages inspection with

```
2021-02-22 18:59:58.416295 | container | Running specname inspection...       pass
2021-02-22 18:59:58.416327 | container | Running modularity inspection...     pass
2021-02-22 18:59:58.417186 | container | Running javabytecode inspection...   pass
2021-02-22 19:00:01.775670 | container | rpminspect: archive_read_next_header(): No such file or directory
2021-02-22 19:00:01.775741 | container | rpminspect: ../lib/inspect_changedfiles.c:228: changedfiles_driver: Assertion `before_uncompressed_file != NULL' failed.
2021-02-22 19:00:01.780042 | container | /usr/bin/rpminspect-fedora: line 3:  2407 Aborted                 /usr/bin/rpminspect -c /usr/share/rpminspect/fedora.yaml $*
2021-02-22 19:00:02.033798 | container | ERROR
2021-02-22 19:00:02.034303 | container | {
2021-02-22 19:00:02.034509 | container |   "delta": "0:00:07.071133",
2021-02-22 19:00:02.034589 | container |   "end": "2021-02-22 19:00:01.781142",
2021-02-22 19:00:02.034654 | container |   "msg": "non-zero return code",
2021-02-22 19:00:02.034716 | container |   "rc": 134,
2021-02-22 19:00:02.034842 | container |   "start": "2021-02-22 18:59:54.710009"
2021-02-22 19:00:02.034906 | container | }
```

See for the full log: https://fedora.softwarefactory-project.io/zuul/build/7440ae99179a4abf9bb057bc4b971383/log/job-output.txt#1190

Any idea about this issue ?

Thanks in advance.

**Comment by dcantrell** [comment]
I am unable to reproduce this problem locally.  A couple of things:

- What version of rpminspect are you using?  I have 1.3.1 built and available in updates-testing, but at least version 1.2 should be used.
- What builds are you comparing here?  Koji told me no such build for grub2-2.04-35.fc35.

Just running the latest stable rpminspect on grub2-2.04-35.fc34 ran fine.  Please advise.

**Comment by morucci** [comment]
I'm able to reproduce out of the CI, locally.

grub2-2.04-35.fc35 was built by the CI from that PR https://src.fedoraproject.org/rpms/grub2/pull-request/13

Basically to reproduce:
```
# I've managed to get this in /var/tmp/rpminspect:

[root@2c9cbe96ae49 zuul-distro-jobs]# find /var/tmp/rpminspect/grub2-2.04-35.fc3*
/var/tmp/rpminspect/grub2-2.04-35.fc34
/var/tmp/rpminspect/grub2-2.04-35.fc34/aarch64
/var/tmp/rpminspect/grub2-2.04-35.fc34/aarch64/grub2-emu-modules-2.04-35.fc34.aarch64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/aarch64/grub2-tools-minimal-2.04-35.fc34.aarch64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/aarch64/grub2-tools-minimal-debuginfo-2.04-35.fc34.aarch64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/aarch64/grub2-efi-aa64-2.04-35.fc34.aarch64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/aarch64/grub2-emu-2.04-35.fc34.aarch64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/aarch64/grub2-tools-2.04-35.fc34.aarch64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/aarch64/grub2-efi-aa64-cdboot-2.04-35.fc34.aarch64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/aarch64/grub2-tools-debuginfo-2.04-35.fc34.aarch64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/aarch64/grub2-debuginfo-2.04-35.fc34.aarch64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/aarch64/grub2-tools-extra-2.04-35.fc34.aarch64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/aarch64/grub2-tools-extra-debuginfo-2.04-35.fc34.aarch64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/aarch64/grub2-debugsource-2.04-35.fc34.aarch64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/aarch64/grub2-emu-debuginfo-2.04-35.fc34.aarch64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/ppc64le
/var/tmp/rpminspect/grub2-2.04-35.fc34/ppc64le/grub2-ppc64le-2.04-35.fc34.ppc64le.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/ppc64le/grub2-tools-extra-2.04-35.fc34.ppc64le.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/ppc64le/grub2-tools-extra-debuginfo-2.04-35.fc34.ppc64le.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/ppc64le/grub2-debugsource-2.04-35.fc34.ppc64le.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/ppc64le/grub2-tools-2.04-35.fc34.ppc64le.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/ppc64le/grub2-tools-debuginfo-2.04-35.fc34.ppc64le.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/ppc64le/grub2-debuginfo-2.04-35.fc34.ppc64le.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/ppc64le/grub2-tools-minimal-2.04-35.fc34.ppc64le.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/ppc64le/grub2-tools-minimal-debuginfo-2.04-35.fc34.ppc64le.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/i686
/var/tmp/rpminspect/grub2-2.04-35.fc34/i686/grub2-debuginfo-2.04-35.fc34.i686.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/i686/grub2-tools-extra-2.04-35.fc34.i686.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/i686/grub2-tools-extra-debuginfo-2.04-35.fc34.i686.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/i686/grub2-pc-2.04-35.fc34.i686.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/i686/grub2-tools-debuginfo-2.04-35.fc34.i686.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/i686/grub2-tools-minimal-2.04-35.fc34.i686.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/i686/grub2-emu-modules-2.04-35.fc34.i686.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/i686/grub2-tools-2.04-35.fc34.i686.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/i686/grub2-emu-debuginfo-2.04-35.fc34.i686.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/i686/grub2-tools-minimal-debuginfo-2.04-35.fc34.i686.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/i686/grub2-debugsource-2.04-35.fc34.i686.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/i686/grub2-emu-2.04-35.fc34.i686.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/src
/var/tmp/rpminspect/grub2-2.04-35.fc34/src/grub2-2.04-35.fc34.src.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/noarch
/var/tmp/rpminspect/grub2-2.04-35.fc34/noarch/grub2-pc-modules-2.04-35.fc34.noarch.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/noarch/grub2-efi-x64-modules-2.04-35.fc34.noarch.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/noarch/grub2-ppc64le-modules-2.04-35.fc34.noarch.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/noarch/grub2-common-2.04-35.fc34.noarch.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/noarch/grub2-efi-arm-modules-2.04-35.fc34.noarch.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/noarch/grub2-efi-ia32-modules-2.04-35.fc34.noarch.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/noarch/grub2-efi-aa64-modules-2.04-35.fc34.noarch.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/armv7hl
/var/tmp/rpminspect/grub2-2.04-35.fc34/armv7hl/grub2-tools-extra-2.04-35.fc34.armv7hl.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/armv7hl/grub2-emu-debuginfo-2.04-35.fc34.armv7hl.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/armv7hl/grub2-tools-extra-debuginfo-2.04-35.fc34.armv7hl.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/armv7hl/grub2-debugsource-2.04-35.fc34.armv7hl.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/armv7hl/grub2-efi-arm-2.04-35.fc34.armv7hl.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/armv7hl/grub2-emu-2.04-35.fc34.armv7hl.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/armv7hl/grub2-tools-2.04-35.fc34.armv7hl.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/armv7hl/grub2-emu-modules-2.04-35.fc34.armv7hl.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/armv7hl/grub2-tools-debuginfo-2.04-35.fc34.armv7hl.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/armv7hl/grub2-efi-arm-cdboot-2.04-35.fc34.armv7hl.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/armv7hl/grub2-debuginfo-2.04-35.fc34.armv7hl.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/armv7hl/grub2-tools-minimal-2.04-35.fc34.armv7hl.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/armv7hl/grub2-tools-minimal-debuginfo-2.04-35.fc34.armv7hl.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-emu-debuginfo-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-tools-efi-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-pc-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-tools-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-tools-extra-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-debuginfo-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-emu-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-tools-minimal-debuginfo-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-efi-x64-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-tools-minimal-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-efi-ia32-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-tools-debuginfo-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-debugsource-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-efi-ia32-cdboot-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-emu-modules-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-tools-extra-debuginfo-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-tools-efi-debuginfo-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc34/x86_64/grub2-efi-x64-cdboot-2.04-35.fc34.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35
/var/tmp/rpminspect/grub2-2.04-35.fc35/src
/var/tmp/rpminspect/grub2-2.04-35.fc35/src/grub2-2.04-35.fc35.src.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/noarch
/var/tmp/rpminspect/grub2-2.04-35.fc35/noarch/grub2-efi-x64-modules-2.04-35.fc35.noarch.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/noarch/grub2-common-2.04-35.fc35.noarch.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/noarch/grub2-efi-ia32-modules-2.04-35.fc35.noarch.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-emu-modules-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-tools-efi-debuginfo-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-tools-extra-debuginfo-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-efi-x64-cdboot-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-emu-debuginfo-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-tools-efi-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-pc-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-tools-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-tools-extra-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-debuginfo-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-tools-minimal-debuginfo-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-emu-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-efi-x64-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-tools-minimal-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-efi-ia32-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-tools-debuginfo-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-debugsource-2.04-35.fc35.x86_64.rpm
/var/tmp/rpminspect/grub2-2.04-35.fc35/x86_64/grub2-efi-ia32-cdboot-2.04-35.fc35.x86_64.rpm

# Then run

rpminspect-fedora -Dv -k -t BAD -a "x86_64,noarch,src" -o /root/zuul-output/logs/result.json -F json /var/tmp/rpminspect/grub2-2.04-35.fc34 /var/tmp/rpminspect/grub2-2.04-35.fc35

...
        origin_prefix_trim:
            - ^(opt/rh/[^/]+/root/)
==========
Running license inspection...        FAIL
Running emptyrpm inspection...       pass
Running lostpayload inspection...    FAIL
Running metadata inspection...       pass
Running manpage inspection...        pass
Running xml inspection...            pass
Running elf inspection...            FAIL
Running desktop inspection...        pass
Running disttag inspection...        pass
Running specname inspection...       pass
Running modularity inspection...     pass
Running javabytecode inspection...   pass
rpminspect: archive_read_next_header(): Success
rpminspect: ../lib/inspect_changedfiles.c:228: changedfiles_driver: Assertion `before_uncompressed_file != NULL' failed.
/usr/bin/rpminspect-fedora: line 3:  6606 Aborted                 (core dumped) /usr/bin/rpminspect -c /usr/share/rpminspect/fedora.yaml $*
```

Here are the files generated from the PR:

"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-2.04-35.fc35.src.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-common-2.04-35.fc35.noarch.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-debuginfo-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-emu-debuginfo-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-emu-modules-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-pc-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-tools-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-tools-debuginfo-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-tools-efi-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-tools-efi-debuginfo-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-tools-extra-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-tools-extra-debuginfo-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-tools-minimal-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-debugsource-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-tools-minimal-debuginfo-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-efi-ia32-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-efi-ia32-cdboot-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-efi-ia32-modules-2.04-35.fc35.noarch.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-efi-x64-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-efi-x64-cdboot-2.04-35.fc35.x86_64.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-efi-x64-modules-2.04-35.fc35.noarch.rpm"
"https://fedora.softwarefactory-project.io/logs/13/13/a32aa179fa4f5d914da176aa0ccceb9b5cd13535/check/rpm-scratch-build/aca3972/repo/grub2-emu-2.04-35.fc35.x86_64.rpm"

I hope It will help.

Thanks

**Comment by dcantrell** [comment]
Thanks for the detailed reproducer instructions.  I was able to recreate this problem just now.  Working on it.

**Comment by dcantrell** [comment]
Figured out the problem.  The uncompress_file() function was not handling compressed empty files correctly.  A zero length file compressed with gzip can be handed to libarchive.  When calling archive_get_next_header(), we actually get ARCHIVE_EOF since we are at the end of the file.  In this case, I modified uncompress_file() to write an empty file so we have something to work with.

Over in inspect_manpage.c, I needed to expand the manpage inspection to handle finding empty man pages.  This was done with a new block that just uses stat(2) on the uncompressed file since we can't rely on the compressed file.  The other checks always pass since handing nothing to libmandoc is "valid" mandoc.

Patch and test cases coming soon.
